src/controller/PromotionController.ts

```diff
@@ -0,0 +1,81 @@
+import { getRepository } from 'typeorm';
+import {
+  Body,
+  Delete,
+  Get,
+  JsonController,
+  Param,
+  Post,
+  QueryParam,
+  OnUndefined,
+  Patch, HttpCode, BadRequestError,
+} from 'routing-controllers';
+import { Promotion } from '../entities/Promotion';
+
+@JsonController()
+export class PromotionController {
+
+  private repository = getRepository(Promotion);
+
+  @Get('/promotions/')
+  async all(@QueryParam('limit') limit: number,
+            @QueryParam('store') store: number,
+            @QueryParam('brand') brand: number,
+            @QueryParam('user') user: number) {
+    const filter: {[key: string]: number}[] = [];
+    if (store) { filter.push({ store }); }
+    if (brand) { filter.push({ brand }); }
+    if (user) { filter.push({ user }); }
+
+    return this.repository.find({
+      relations: ['store', 'brand', 'user'],
+      where: filter,
+      take: limit,
+    });
+  }
+
+  @Get('/promotions/:id')
+  async one(@Param('id') id: number) {
+    return this.repository.findOne({ id }, {
+      relations: ['store', 'brand', 'user'],
+    });
+  }
+
+  @Get('/promotions/for/:userId')
+  async allForUser(@Param('userId') userId: number) {
+    return this.repository.find({ userId });
```

Comment:
Is this really a relevant endpoint ? What's the use case ?

Reply:
To get all promotions poster by a user I asked adrien and he need it

Reply:
Why not /promotions/?user=userId ?